cmd/dep/govendor_importer.go

```diff
@@ -0,0 +1,148 @@
+// Copyright 2016 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+
+	"github.com/golang/dep"
+	fb "github.com/golang/dep/internal/feedback"
+	"github.com/golang/dep/internal/gps"
+	"github.com/pkg/errors"
+)
+
+const govendorDir = "vendor"
+const govendorName = "vendor.json"
+
+type govendorImporter struct {
+	file govendorFile
+
+	logger  *log.Logger
+	verbose bool
+	sm      gps.SourceManager
+}
+
+func newGovendorImporter(logger *log.Logger, verbose bool, sm gps.SourceManager) *govendorImporter {
+	return &govendorImporter{
+		logger:  logger,
+		verbose: verbose,
+		sm:      sm,
+	}
+}
+
+// File is the structure of the vendor file.
+type govendorFile struct {
+	RootPath string // Import path of vendor folder
+
+	Comment string
+
+	Ignore string
+
+	Package []*govendorPackage
+}
+
+// Package represents each package.
+type govendorPackage struct {
+	// If delete is set to true the package will not be written to the vendor file.
+	Remove bool
+
+	// If new is set to true the package will be treated as a new package to the file.
+	Add bool
```

Comment:
Are `Remove` and `Add` out of the usual govendor spec? Just curious how they would work 🙂